pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -975,10 +951,11 @@ func (nc *Controller) tryUpdateNodeHealth(node *v1.Node) (time.Duration, v1.Node
 				}
 			}
 		}
+		// We need to update currentReadyCondition due to its value potentially changed.
+		_, currentReadyCondition = nodeutil.GetNodeCondition(&node.Status, v1.NodeReady)
```

Comment:
If removing if didn't cause any tests to fail, it seems quite clear that some unit test is missing - please add test case to excercise this case.

Reply:
I've added the tests that checks if returned currentReadyCondition is the same as one calculated based on the stored node status. IMHO it should cover all of the cases: all statuses, nil status and status changed to unknown.